File: cpp/42a_1368A_C+=.cpp
#include<bits/stdc++.h>

int main(){
    int t, a, b, n, cnt, temp1, temp2;
    std::vector<int> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>temp1>>temp2>>n;
        cnt = 0, a = std::min(temp1, temp2), b = std::max(temp1, temp2);
        while (a <= n && b <= n){
            if (cnt % 2 == 0){
                a += b;
            }
            else{
                b += a;
            }
            cnt ++;
        }
        ans.push_back(cnt);
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

//3412102 17440 26861982 == 5
//15 12409 160767 == 6

File: python/42a_1368A_C+=.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        a, b, n = input().split()
        a, b, n = int(a), int(b), int(n)
        cnt = 0
        a, b = min(a,b), max(a,b)
        while a <= n and b <= n:
            if cnt % 2 == 0:
                a += b
            else:
                b += a
            cnt += 1
        ans.append(cnt)
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()